### ISSUE TITLE
Revise Input Mapping Data Structure

### ISSUE THREAD

**Issue by MirisWisdom** [HaloSPV3/HCE#102]
At the moment, a `Dictionary<Profile.ProfileInput.Action, Profile.ProfileInput.Button>` is used to represent the input mappings, i.e. what button corresponds to what action. This limits us from using `Profile.ProfileInput.Action` members with shared values; dictionary objects expect an unique key for each entry.

To resolve this, we could introduce a `Binding` class with two properties: `Button` and `Action`. A `List<Binding>` collection can represent the entire input-action mapping:

```cs
class Binding
{
   public Button Button { get; set; }
   public Action Action { get; set; }
}
```
```cs
var mappings = new List<Binding>
{
   new Binding { Button = ANALOGUE_UP,   Action = WALK_FORWARD   },
   new Binding { Button = ANALOGUE_DOWN, Action = WALK_BACKWARDS },
};
```

**Comment by BinToss** [comment]
I still have that branch available in case we pick this up later.

**Comment by BinToss** [comment]
It seems replacing the Dictionaries with Lists is unnecessary.
The Menu bindings are the only exceptions the the action-input bindings rule and the profile code has been adjusted to account for this.
Resolved by #206
